google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/AppEngineCloudSdkService.java

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.sdk;
+
+import com.google.cloud.tools.appengine.api.whitelist.AppEngineJreWhitelist;
+import com.google.cloud.tools.intellij.appengine.jps.model.impl.JpsAppEngineModuleExtensionImpl;
+
+import com.intellij.execution.configurations.ParametersList;
+import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.util.io.FileUtil;
+import com.intellij.openapi.util.io.JarUtil;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.openapi.vfs.VfsUtil;
+import com.intellij.openapi.vfs.VfsUtilCore;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.openapi.vfs.VirtualFileManager;
+import com.intellij.util.containers.ContainerUtil;
+
+import gnu.trove.THashMap;
+import gnu.trove.THashSet;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.jar.Attributes;
+
+/**
+ * Default implementation of {@link AppEngineSdkService} backed by {@link PropertiesComponent} for
+ * serialization.
+ *
+ */
+// TODO Offload various path logic for retrieving AE libs to the common library once implemented
+public class AppEngineCloudSdkService extends AppEngineSdkService {
+
+  private static final Logger logger = Logger.getInstance(AppEngineCloudSdkService.class);
+
+  private PropertiesComponent propertiesComponent;
+  private static final String CLOUD_SDK_PROPERTY_KEY = "GCT_CLOUD_SDK_HOME_PATH";
+  private static final Path JAVA_TOOLS_RELATIVE_PATH
+      = Paths.get("platform", "google_appengine", "google", "appengine", "tools", "java");
+  private Map<String, Set<String>> myMethodsBlackList;
+
+  public AppEngineCloudSdkService() {
+    this.propertiesComponent = PropertiesComponent.getInstance();
```

Comment:
You could initialize propertiesComponent at declaration time, since you're not assigning anywhere else.